src/configure

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/env gsi-script
+
+(define features
+  '((libxml  . #f)
+    (libyaml . #f)
+    (zlib    . #t)
+    (sqlite  . #t)
+    (mysql   . #f)
+    (lmdb    . #f)
+    (leveldb . #f)))
+
+(define install-prefix #f)
+(define gambit-prefix #f)
+
+(define files-to-patch
+  '("gerbil/compiler/driver.ss"
+    "gerbil/boot/gx-init.scm"
+    "gerbil/boot/gx-init-exe.scm"
+    "gerbil/gxc"
+    "gerbil/gxi"
+    "gerbil/gxi-build-script"
+    "std/build-features.ss"))
+
+(define (string-starts-with? s prefix)
+  (let* ((prefix-len (string-length prefix)))
+    (and (<= prefix-len (string-length s))
+         (string=? prefix (substring s 0 prefix-len)))))
+
+(define (string-contains? s find)
+  (let ((find-len (string-length find)))
+    (let loop ((i 0))
+      (cond
+       ((< (string-length s) (+ i find-len))
+        #f)
+       ((string=? find (substring s i (+ i find-len))))
+       (else
+        (loop (+ 1 i)))))))
+
+(define (feedback-mid . message-bits)
+  (display ">>> ")
+  (for-each display message-bits)
+  (newline))
+
+(define (feedback-low . message-bits)
+  (display "[*] ")
+  (for-each display message-bits)
+  (newline))
+
+(define-record-type :patch
+  (make-patch replacement prefix substrings)
+  patch?
+  (replacement patch-replacement)
+  (prefix      patch-prefix)
+  (substrings  patch-substrings))
+
+(define (patch-prefix-matches-line? patch line)
+  (string-starts-with? line (patch-prefix patch)))
+
+(define (patch-substrings-match-line? patch line)
+  (fold (lambda (substring answer)
+          (and answer (string-contains? line substring)))
+        #t
+        (patch-substrings patch)))
+
+(define (patch-matches-line? patch line)
+  (and (patch-prefix-matches-line? patch line)
+       (patch-substrings-match-line? patch line)))
+
+(define (patch-apply patch line)
+  (if (patch-matches-line? patch line)
+    (patch-replacement patch)
+    line))
+
+(define (die . message-bits)
+  (parameterize ((current-output-port (current-error-port)))
+    (display "*** ERROR; ")
+    (for-each display message-bits)
+    (newline))
+  (exit 1))
+
+(define (enable-feature! feature value)
+  (let ((cell (assq (string->symbol feature) features)))
+    (if (not cell)
```

Comment:
Nit: I prefer `when` and `unless` instead of one-armed `if`.

Reply:
me too, but older gambits don't have it!

Reply:
Ouch, `when` only appeared in v4.9.2, and Gerbil extends backward compatibility to v4.9.0.